packages/myst-to-react/src/code.tsx

```diff
@@ -32,8 +33,9 @@ function normalizeLanguage(lang?: string): string | undefined {
   }
 }
 
+type HighlightPropsType = ComponentProps<typeof SyntaxHighlighter>;
+
 export function CodeBlock(props: Props) {
-  const { isLight } = useThemeSwitcher();
```

Comment:
Is there a way to make this work?

I don't love having two of every code element in the DOM (and all of the additional nodes for syntax highlighting). Especially for notebooks. In many of the existing sites, we are all ready getting lighthouse warnings that there are many DOM nodes and that starts to kill performance.

Perhaps re-trigger this in the static builds, and live with the flicker for code.

Reply:
Yeah, I don't love the way it's needed either.

I think it should be possible to return a single DOM node when SSR if the cookie is set. That only helps subsequent visits.

What we probably need to do is look into pushing this state into CSS variables. We already do this for our tailwind styling, but I didn't immediately see a way to do it here. Perhaps we need a custom theme for the code block.